tests/h/search/query_test.py

```diff
@@ -141,12 +141,88 @@ def search(self, search):
 
 
 class TestUriFilter(object):
+    @pytest.mark.parametrize("field", ("uri", "url"))
+    def test_filters_by_field(self, search, Annotation, field):
+        Annotation(target_uri="https://foo.com")
+        expected_ids = [Annotation(target_uri="https://bar.com").id]
+
+        result = search.run({field: "https://bar.com"})
+
+        assert sorted(result.annotation_ids) == sorted(expected_ids)
+
+    def test_filters_on_whole_url(self, search, Annotation):
+        Annotation(target_uri="http://bar.com/foo")
+        expected_ids = [Annotation(target_uri="http://bar.com").id,
+                        Annotation(target_uri="http://bar.com/").id]
+
+        result = search.run({"url": "http://bar.com"})
+
+        assert sorted(result.annotation_ids) == sorted(expected_ids)
+
+    def test_filter_matches_buggy_uri(self, search, Annotation):
```

Comment:
It would be more accurate to describe this as an _invalid_ URI rather than a "buggy" one. It is the _buggy_ client that creates an _invalid URI_ 😜